lambda/matching.ml

```diff
@@ -667,22 +665,69 @@ end = struct
     | [] -> default
     | _ -> (matrix, raise_num) :: default
 
-  let specialize_matrix matcher pss =
+  let specialize_matrix arity matcher pss =
     let rec filter_rec = function
       | (p :: ps) :: rem -> (
           match p.pat_desc with
           | Tpat_alias (p, _, _) -> filter_rec ((p :: ps) :: rem)
           | Tpat_var _ -> filter_rec ((omega :: ps) :: rem)
+          | Tpat_or (p1, p2, _) -> (
```

Comment:
@Octachron suggests moving this large and tricky part as a separate mutually-recursive function. Will do.

Reply:
I refactored with an auxiliary function as suggested, and in fact added another small auxiliary function and some additional explanatory comments.